app/Controllers/Http/ParticipationImagesController.ts

```diff
@@ -0,0 +1,72 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import ParticipationImage from 'App/Models/ParticipationImage'
+import SetSubmission from 'App/Models/SetSubmission'
+import BaseController from './BaseController'
+import Account from 'App/Models/Account'
+import { isAdmin } from 'App/Middleware/AdminAuth'
+import { DateTime } from 'luxon'
+import NotAuthenticated from 'App/Exceptions/NotAuthenticated'
+import NotAuthorized from 'App/Exceptions/NotAuthorized'
+import BadRequest from 'App/Exceptions/BadRequest'
+import Image from 'App/Models/Image'
+
+export default class ParticipationImagesController extends BaseController {
+  public async store({ request, session }: HttpContextContract) {
+    const address = session.get('siwe')?.address?.toLowerCase()
+    if (!address) throw new NotAuthenticated()
+
+    const submissionId: string | undefined = request.input('submissionId')
+    const imageIds: string[] | undefined = request.input('imageIds')
+    
+    if (!submissionId) throw new BadRequest('submissionId is missing')
+    if (!imageIds?.length) throw new BadRequest('imageIds array is missing or empty')
+
+    const submission = await SetSubmission.query().where('uuid', submissionId).first()
+    if (!submission) throw new BadRequest('Submission not found')
+    if (!submission.openForParticipation) throw new BadRequest('This set is not open for participation')
+
+    await Account.firstOrCreate({ address })
+
+    const images = await Image.query().whereIn('uuid', imageIds)
+    if (images.length !== imageIds.length) throw new BadRequest('No image not found')
+
+    if (images.some(img => img.creator !== address)) {
+      throw new NotAuthorized('You can only use images you uploaded')
```

Comment:
Could add this directly to the query instead of doing in memory... 

`.where('creator', address)`